src/GameEngine/GameEngineDriver.cpp

```diff
@@ -0,0 +1,52 @@
+//
+// Created by ker10 on 2019-10-23.
+//
+#include <vector>
+#include <string>
+#include <cassert>
+#include "GameEngineDriver.h"
+#include "GameEngine.h"
+#include "../Terminal/Terminal.h"
+
+using namespace std;
+
+namespace GameEngine {
+    namespace Driver {
+
+        bool run(){
+            bool passed =   map_loaded() &&
+                            right_amount_of_player() &&
+                            right_amount_of_card_in_deck();
+
+            return passed;
+        }
+
```

Comment:
For this case, would bit be better to have an initializing function, where we store the GameEngine instance? As well as comments stating the start_test, and test_reset reasons

Reply:
There’s always only one gameengine instance. It’s a singleton. You can’t call new on it. You can instance() and it returns the only gameengine there is. 

Start test/ test reset are basically hacks to mock a game